### PR TITLE
[IMP] base: increase password work factor

### PR DESCRIPTION
**Pull request by xmo-odoo**
And make configurable via an ICP. Also modernize passlib usage:

* Remove global `DEFAULT_CRYPT_CONTEXT`, create inline (as overhead
  should not be too huge given what we're doing with it), and
  `ormcache()` for safety, the caches should be invalidated on any ICP
  addition, removal, or update, so user update to the rounds
  configuration should get reflected immediately.
* Switch on `deprecated=["auto"]`, feature was added in 1.6 and we now
  depend on 1.7.
* Remove mentions of `encrypt`, it is deprecated in 1.7.

### CODE REVIEW

File: odoo/addons/base/models/res_users.py [hunk 1]
```diff
@@ -919,15 +905,27 @@ def _is_superuser(self):
...
+        return CryptContext(
+            # kdf which can be verified by the context. The default encryption
+            # kdf is the first of the list
+            # TODO: provide key to *set* the alg?
+            ['pbkdf2_sha512', 'plaintext'],
+            # deprecated algorithms are still verified as usual, but
+            # ``needs_update`` will indicate that the stored hash should be
+            # replaced by a more recent algorithm.
+            deprecated=['auto'],
+            pbkdf2_sha512__rounds=int(cfg.get_param('password.hashing.rounds', 350000)),
```
Comment (odony) [inline odoo/addons/base/models/res_users.py:927]: should we ignore the ICP if it's set below the default number of rounds? some people might think that 999 is a huge number of rounds, for example :thinking: 
(obviously they shouldn't be changing the ICP if they don't know what they're doing, but you know people...)

Comment (xmo-odoo) [reply odoo/addons/base/models/res_users.py]: Something like that?

Comment (xmo-odoo) [reply odoo/addons/base/models/res_users.py]: Could default to 0 now that I think about it, no need to bother repeating `MIN_ROUNDS`

**Review by odony — APPROVED** [review]

**Comment by xmo-odoo** [comment]
@robodoo r+

**Pull request closed without merging**